Allow JIT compilation for PPC targets (e.g. "pwr9" CPUs)
From https://github.com/tensorflow/tensorflow/pull/55306
Backported for TF 2.7.1

Author: Alexander Grund (TU Dresden)

diff --git a/third_party/cpuinfo/BUILD.bazel b/third_party/cpuinfo/BUILD.bazel
index eb2937d20ef..dfea408db94 100644
--- a/third_party/cpuinfo/BUILD.bazel
+++ b/third_party/cpuinfo/BUILD.bazel
@@ -109,6 +109,7 @@ cc_library(
         ":linux_mips64": COMMON_SRCS + LINUX_SRCS,
         ":linux_riscv64": COMMON_SRCS + LINUX_SRCS,
         ":linux_s390x": COMMON_SRCS + LINUX_SRCS,
+        ":linux_ppc64le": COMMON_SRCS + LINUX_SRCS,
         ":macos_x86_64": COMMON_SRCS + X86_SRCS + MACH_SRCS + MACH_X86_SRCS,
         ":macos_arm64": COMMON_SRCS + MACH_SRCS + MACH_ARM_SRCS,
         ":windows_x86_64": COMMON_SRCS + X86_SRCS + WINDOWS_X86_SRCS,
@@ -232,6 +233,11 @@ config_setting(
     values = {"cpu": "s390x"},
 )
 
+config_setting(
+    name = "linux_ppc64le",
+    values = {"cpu": "ppc"},
+)
+
 config_setting(
     name = "macos_x86_64",
     values = {
diff --git a/third_party/llvm/macos_build_fix.patch b/third_party/llvm/macos_build_fix.patch
index 8ff52f906a6..4467081c410 100644
--- a/third_party/llvm/macos_build_fix.patch
+++ b/third_party/llvm/macos_build_fix.patch
@@ -22,6 +22,48 @@ index ff64df694048..55f3077bff74 100644
      "@bazel_tools//src/conditions:linux_aarch64": native_arch_defines("AArch64", "aarch64-unknown-linux-gnu"),
      "//conditions:default": native_arch_defines("X86", "x86_64-unknown-linux-gnu"),
  }) + [
+
+
+From b250c34bbd415b2a8c3e0532e98591ab1780cda6 Mon Sep 17 00:00:00 2001
+From: Nishidha Panpaliya <npanpa23@in.ibm.com>
+Date: Mon, 21 Mar 2022 09:51:36 -0400
+Subject: [PATCH] Fix for ppc64le
+
+---
+ utils/bazel/llvm-project-overlay/llvm/config.bzl | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/utils/bazel/llvm-project-overlay/llvm/config.bzl b/utils/bazel/llvm-project-overlay/llvm/config.bzl
+index 772714f38941..9ed63e8d44a3 100644
+--- a/utils/bazel/llvm-project-overlay/llvm/config.bzl
++++ b/utils/bazel/llvm-project-overlay/llvm/config.bzl
+@@ -78,6 +78,7 @@
+     "@bazel_tools//src/conditions:windows": native_arch_defines("X86", "x86_64-pc-win32"),
+     "@bazel_tools//src/conditions:darwin": native_arch_defines("X86", "x86_64-unknown-darwin"),
+     "@bazel_tools//src/conditions:linux_aarch64": native_arch_defines("AArch64", "aarch64-unknown-linux-gnu"),
++    "@bazel_tools//src/conditions:linux_ppc64le": native_arch_defines("PowerPC", "powerpc64le-unknown-linux-gnu"),
+     "//conditions:default": native_arch_defines("X86", "x86_64-unknown-linux-gnu"),
+ }) + [
+     # These shouldn't be needed by the C++11 standard, but are for some
+
+---
+Fix link errors on PPC caused by above patch: https://github.com/llvm/llvm-project/issues/59590
+
+Author: Alexander Grund (TU Dresden)
+
+diff --git a/utils/bazel/llvm-project-overlay/mlir/BUILD.bazel b/utils/bazel/llvm-project-overlay/mlir/BUILD.bazel
+index f4dfb3791477..da92b53b7a2b 100644
+--- a/utils/bazel/llvm-project-overlay/mlir/BUILD.bazel
++++ b/utils/bazel/llvm-project-overlay/mlir/BUILD.bazel
+@@ -4986,6 +4986,7 @@ cc_library(
+         ":Support",
+         ":ToLLVMIRTranslation",
+         ":Translation",
++        "//llvm:AllTargetsAsmParsers",
+         "//llvm:BitReader",
+         "//llvm:BitWriter",
+         "//llvm:Core",
+
 -- 
 2.33.0.685.g46640cef36-goog
